src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java

```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String studentID = Optional.ofNullable((String)request.getSession(false).getAttribute("userId")).orElse("-1");
         long id = Long.parseLong(request.getParameter("id"));
 
+        TutorSession tutorSession = datastore.getScheduledSession(id);
+
+        System.out.println(tutorSession.getStudentID());
+
+        Entity studentEntity = datastore.getStudentForUserId(tutorSession.getStudentID());
+        Entity tutorEntity = datastore.getTutorForUserId(tutorSession.getTutorID());
+        TimeRange timeslot = tutorSession.getTimeslot();
+
+        boolean testTutorEmail = sendCancellationEmailToTutor(studentEntity, tutorEntity, timeslot);
+        boolean testStudentEmail = sendCancellationEmailToStudent(studentEntity, tutorEntity, timeslot);
+
         // Remove tutor session
         datastore.deleteTutorSession(studentID, id);
 
         String json = new Gson().toJson(datastore.getScheduledSession(id));
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(json + "testTutorEmail: " + testTutorEmail + " testStudentEmail: " + testStudentEmail);
         return;
     }
+
+    public boolean sendCancellationEmailToTutor(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
+
+        String to = (String) tutorEntity.getProperty("email");
+        String subject = "Cancelled Tutoring Session";
+        String studentName = (String) studentEntity.getProperty("name");
+        Calendar date = timeslot.getDate();
+        String month = new DateFormatSymbols().getMonths()[date.get(Calendar.MONTH)];
+        int day = date.get(Calendar.DAY_OF_MONTH);
+        int year = date.get(Calendar.YEAR);
+        String message = studentName + " has cancelled their tutoring session with you on " +
+                        month + " " + day + ", " + year + ". Check your My Student page for more information.";
+
+        try {
+            Message msg = new MimeMessage(session);
+            msg.setFrom(new InternetAddress("contact@icecube-step-2020.appspotmail.com", "Sullivan"));
+            msg.addRecipient(Message.RecipientType.TO, new InternetAddress(to));
+            msg.setSubject(subject);
+            msg.setText(message);
+            Transport.send(msg);
+            return true;
+        } catch (AddressException e) {
+            System.out.println("Failed to set email address.");
+            return false;
+        } catch (MessagingException e) {
+            System.out.println("Failed to send email.");
+            return false;
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Failed to encode email.");
+            return false;
+        }
+    }
+
+    public boolean sendCancellationEmailToStudent(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
```

Comment:
Do these methods need to be public?

Reply:
Not at all, I'll make them private. Thanks!